appIcons.js

```diff
@@ -247,38 +266,39 @@ const MyAppIcon = new Lang.Class({
                 return;
 
             }
-            else if (this._dtdSettings.get_boolean('minimize-shift') && modifiers & Clutter.ModifierType.SHIFT_MASK) {
+            else if (_clickAction == clickAction.MINIMIZE
+                     && (modifiers & Clutter.ModifierType.SHIFT_MASK || button == 2)) {
                 // On double click, minimize all windows in the current workspace
                 minimizeWindow(this.app, event.get_click_count() > 1, this._dtdSettings);
             }
             else if (this.app == focusedApp && !Main.overview._shown) {
-                if (this._dtdSettings.get_enum('click-action') == clickAction.CYCLE_WINDOWS)
+                if (_clickAction == clickAction.CYCLE_WINDOWS)
                     cycleThroughWindows(this.app, this._dtdSettings);
-                else if (this._dtdSettings.get_enum('click-action') == clickAction.MINIMIZE)
+                else if (_clickAction == clickAction.MINIMIZE)
                     minimizeWindow(this.app, true, this._dtdSettings);
-                else if (this._dtdSettings.get_enum('click-action') == clickAction.LAUNCH)
+                else if (_clickAction == clickAction.LAUNCH)
                     this.app.open_new_window(-1);
+                else if (_clickAction == clickAction.QUIT)
+                    closeAllWindows(this.app, this._dtdSettings);
             }
             else {
                 // Activate all window of the app or only le last used
-                if (this._dtdSettings.get_enum('click-action') == clickAction.CYCLE_WINDOWS && !Main.overview._shown) {
+                if (_clickAction == clickAction.CYCLE_WINDOWS && !Main.overview._shown) {
                     // If click cycles through windows I can activate one windows at a time
                     // However, when using isolation, we need to open a new
                     // window if there are no windows in the current WS
                     let windows = getInterestingWindows(this.app, this._dtdSettings);
-                    if (windows.length == 0)
```

Comment:
I had added these lines when implementing workspace isolation.
However, we now check that windows.length > 0 in the (new) line 260, so this is not necessary anymore.